server/cluster/cluster.go

```diff
@@ -1021,30 +1021,34 @@ func (c *RaftCluster) RemoveStore(storeID uint64) error {
 	}
 
 	// Remove an offline store should be OK, nothing to do.
-	if store.IsOffline() {
+	if store.IsOffline() && store.IsPhysicallyDestroyed() == physicallyDestroyed {
```

Comment:
Do we need to compare `physicallyDestroyed` here ? i think this is unnecessary.

Reply:
we needn't get the lock and set the status again when the status is already as we expected. By the way, it is possible a store may call the API with the following steps:
1. offline the store normally   DELETE store?force=false
2. Find the store is physically destroyed, do not want it to start again: DELETE store?force=true

Reply:
> Find the store is physically destroyed, do not want it to start again: DELETE store?force=true

Why we need to forbid it? I think calling the api `delete` with force=false , then `force=true` is common to see.

Reply:
If there is no judgment here, first `delete && force=false` and then `delete && force=true`, the latter will not be able to mark the `physicallyDestroyed` correctly. I think it needs to compare `physicallyDestroyed`.

Reply:
Thanks for clarifying